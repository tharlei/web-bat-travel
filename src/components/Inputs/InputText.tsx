interface InputTextProps {
  setValue: (value: string) => void;
  label: string;
}

export function InputText({ label, setValue }: InputTextProps) {
  return (
    <div className="flex flex-col">
      <label className="uppercase text-gray-900 text-xs -mb-6 ml-2 z-10">
        {label}
      </label>
      <input
        required
        type="text"
        onChange={(event) => setValue(event.target.value)}
        className="bg-gray-100 rounded border-0 pt-8 "
      />
    </div>
  );
}
