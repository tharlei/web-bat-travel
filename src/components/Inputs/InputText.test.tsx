import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { InputText } from './InputText';

describe('InputText component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <InputText setValue={(value) => value} label="Test" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
