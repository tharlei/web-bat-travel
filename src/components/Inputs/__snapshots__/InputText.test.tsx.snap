// Vitest Snapshot v1

exports[`InputText component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <label
      class="uppercase text-gray-900 text-xs -mb-6 ml-2 z-10"
    >
      Test
    </label>
    <input
      class="bg-gray-100 rounded border-0 pt-8 "
      required=""
      type="text"
    />
  </div>
</DocumentFragment>
`;
